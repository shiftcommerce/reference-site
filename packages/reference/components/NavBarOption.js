import { Component } from 'react'
import Link from 'next/link'
import classNames from 'classnames'

class NavBarOption extends Component {
  render() {
    let {
      link,
      title,
      index
    } = this.props

    return <Link href={ link }>
      <a className={ classNames('o-nav__option', { 'o-nav__option--first': index === 0 }) } >
        { title }
      </a>
    </Link>
  }
}

export default NavBarOption