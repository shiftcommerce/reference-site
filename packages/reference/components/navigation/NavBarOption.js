// Libraries
import { Component } from 'react'
import Link from 'next/link'
import classNames from 'classnames'

class NavBarOption extends Component {
  render () {
    let {
      href,
      as,
      title,
      index
    } = this.props

    return (
      <Link href={href} as={as}>
        <a className={classNames('o-nav__option', { 'o-nav__option--first': index === 0 })} >
          { title }
        </a>
      </Link>
    )
  }
}

export default NavBarOption
