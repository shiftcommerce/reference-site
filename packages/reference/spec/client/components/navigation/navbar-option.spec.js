// Libraries
import Link from 'next/link'

// Components
import NavBarOption from '../../../../client/components/navigation/navbar-option'

test('renders the links', () => {
  // Arrange

  // Act
  const wrapper = shallow(
    <NavBarOption title='Mens' href='/categories/mens' as='/categories/mens' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toContainReact(<Link href="/slug?slug=/categories/mens" as="/categories/mens"><a className="c-nav__option"><div className="c-nav__option-label">Mens</div><div className="c-nav__option-text">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div></a></Link>)
})
