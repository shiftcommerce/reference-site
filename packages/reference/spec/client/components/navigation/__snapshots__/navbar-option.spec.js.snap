// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the links 1`] = `
<Link
  as="/categories/mens"
  href="/slug?slug=/categories/mens"
>
  <a
    className="c-nav__option"
  >
    <div
      className="c-nav__option-label"
    >
      Mens
    </div>
    <div
      className="c-nav__option-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>
  </a>
</Link>
`;
